Added several addresses CRUD and added addresses to command "all"

KostiantynMoskalenko opened the pull request:
Added several addresses CRUD and added addresses to command "all"
Created functionality for following commands: "add-address",  "edit-address",  "address" and  "remove-address". 
Added functionality to command "all" - added addresses to show.
![Commands with address3](https://github.com/CarpathianUA/bot-assistant/assets/143954608/34308ed8-7193-4a4d-a7c8-36fc088b1c69)

KostiantynMoskalenko committed:
commit

KostiantynMoskalenko committed:
Update contact_handlers.py removed unexpected 4 spaces

Remove 4 spaces on the string 119

CarpathianUA commented on .gitignore:
```diff
@@ -167,3 +167,7 @@ cython_debug/
+
+# Windows venv folder/files
+Scripts
+pyvenv.cfg
```
@KostiantynMoskalenko  Please fix empty new lines at the enf of file. Each file should have a new line at the end (you can also configure it in VScode to add new line of file save)

KostiantynMoskalenko replied:
What do you mean "new line"? What is it? What should I do? As far as I can see every file already has empty line in the end? BTW is it requirement of PEP8? Or something another?

CarpathianUA replied:
new line at the end of file

CarpathianUA replied:
it's already has because I've added it during conflicts resolution

CarpathianUA commented on modules/bot_assistant/constants/commands.py:
```diff
@@ -8,6 +8,10 @@
     "remove-phone": contact_handlers.remove_phone,
+    "add-address": contact_handlers.add_address,
+    "edit-address": contact_handlers.edit_address,
+    "address": contact_handlers.get_contact_address,
```
@KostiantynMoskalenko Recently naming convention changed, we define shows as `show-<entity>` now. E.g. `show-birthday`, `show-email`. Please update you functionality accordingly, thanks

CarpathianUA commented on modules/bot_assistant/constants/commands.py:
```diff
@@ -39,6 +43,18 @@
+        "Add a contact's address",
+        "Example: add-address AlanWake str.Sevchenko,18/5",
+    ),
+    "change-address": (
```
Same here, please change to `edit-address`

CarpathianUA committed:
Merge branch 'master' into feat/km/addresses

KostiantynMoskalenko committed:
Update commands.py changed address to show-address and change-address to edit-address

Update commands.py changed command address to show-address and change-address to edit-address

KostiantynMoskalenko committed:
Merge branch 'master' into feat/km/addresses

KostiantynMoskalenko committed:
commit

KostiantynMoskalenko committed:
Merge branch 'feat/km/addresses' of https://github.com/CarpathianUA/bot-assistant into feat/km/addresses

KostiantynMoskalenko committed:
commit

KostiantynMoskalenko committed:
commit

KostiantynMoskalenko committed:
commit

CarpathianUA commented on modules/bot_assistant/handlers/contact_handlers.py:
```diff
@@ -101,6 +101,57 @@ def remove_phone(args, address_book):
+        raise exceptions.InvalidArgsError
+    name, address = args
+    record = address_book.data.get(name)
+    if not record:
```
@KostiantynMoskalenko here we check if `name` exists in address book, since `record = address_book.data.get(name)`. We need to raise ContactDoesntExist error here.
